kratos/processes/apply_periodic_boundary_condition_process.h

```diff
@@ -0,0 +1,196 @@
+     * @param rSalveNode The slave node which is to be connected to the rHostGeometry.
+     * @param rHostedGeometry the Host geometry which has the rSlaveNode.
+     * @param rWeights The weights with which the rSlaveNode is connected to the rHostedGeometry's nodes.
+     * @param rVarName The name of the vector variable on which periodic boundary condition can be applied.
+     */
+    template <int TDim>
+    void ConstraintSlaveNodeWithConditionForVectorVariable(NodeType& rSalveNode, const GeometryType& rHostedGeometry, const VectorType& rWeights, const std::string& rVarName );
+
+    /**
+     * @brief   The function add the linear master-slave constraint to rModelPart. This function is specifically for applying
+     *          periodic conditions for scalar variable. This distinction is because, for scalar variables the variable need NOT be
+     *          transformed according to the transfromation specified in the settings.
+     * @param rSalveNode The slave node which is to be connected to the rHostGeometry.
+     * @param rHostedGeometry the Host geometry which has the rSlaveNode.
+     * @param rWeights The weights with which the rSlaveNode is connected to the rHostedGeometry's nodes.
+     * @param rVarName The name of the scalar variable on which periodic boundary condition can be applied.
+     */
+    template <int TDim>
+    void ConstraintSlaveNodeWithConditionForScalarVariable(NodeType& rSalveNode, const GeometryType& rHostedGeometry, const VectorType& rWeights, const std::string& rVarName );
+
+    /**
+     * @brief   Fuctions calculates the transformation matrix to account for the moving the two periodic condition modelparts together.
+     */
+    void CalculateTransformationMatrix();
+
+    /**
+     * @brief   Calculates the transformation matrix which translates the given vector alone mDirOfTranslation by mModulus
+     * @param   Modulus is the magnitude by which the translation should happen in the direction of mDirOfTranslation.
+     * @param   rMatrix is the transformation matrix which will be calculated in this function. This should be of correct size (4x4).
+     */
+    void CalculateTranslationMatrix(const double Modulus, MatrixType& rMatrix);
+
+    /**
+     * @brief   Calculates the transformation matrix which rotates the given vector around mAxisOfRotationVector and mCenterOfRotation
+     *          by provided Theta and stores the result in rMatrix The following code is generated from MATLAB and is adapted here.
+     * @param   Theta is the angle of rotation about mAxisOfRotationVector and mCenterOfRotation.
+     * @param   rMatrix is the transformation matrix which will be calculated in this function. This should be of correct size (4x4).
+     */
+    void CalculateRotationMatrix(const double Theta, MatrixType& rMatrix );
+
+    /*
+     * @brief Rotates a given point(node_cords) in space around a given mAxisOfRotationVector by an angle thetha
```

Comment:
There is nothing in the function name which suggests rotation. A carefully named function is better than a comment explaining it. It should also go with the rotation data.